### PR TITLE
feat: add writing of cache file and loading it

### LINKED ISSUE #90: [Feature] Persisting between full webpack builds
You do have `copyUnmodified` set to `false` as default, but that cache is never stored anywhere and thus fails to actually cache anything

### PR DESCRIPTION
**Pull request by deiga**
### Notable Changes
---

Only problem here, is that most tests fail, because we skip files and they are not addded to compilation assets.

A solution would be to change the logic here, so that files are always added to compilation assets.

Another solution could be to skip writing cache files and just compare `from` and `to` files with each other to see if there are cahnges

### Issues
---

- Fixes #90 

> ℹ️  edited by @michael-ciniawsky (Formatting)

### CODE REVIEW

**Comment by michael-ciniawsky** [comment]
@deiga Could you please rebase against current `master` ? :)

File: package.json [hunk 1]
```diff
@@ -24,6 +16,16 @@
     "minimatch": "^3.0.4",
     "node-dir": "^0.1.10"
   },
+  "homepage": "https://github.com/webpack-contrib/copy-webpack-plugin",
```
Comment (evilebottnawi) [inline package.json]: Don't change anything in `package.json`, only add needed package(s), thank you!

File: src/index.js [hunk 2]
```diff
@@ -131,4 +149,4 @@ function CopyWebpackPlugin(patterns = [], options = {}) {
 }
 
 CopyWebpackPlugin['default'] = CopyWebpackPlugin;
-module.exports = CopyWebpackPlugin;
+module.exports = CopyWebpackPlugin;
```
Comment (evilebottnawi) [inline src/index.js]: Don't do this

File: src/writeFile.js [hunk 3]
```diff
@@ -73,6 +73,9 @@ export default function writeFile(globalRef, pattern, file) {
                 written[file.absoluteFrom] = {
                     [hash]: true
                 };
+                if (!copyUnmodified) {
+                    fs.writeFileSync(cachePath, JSON.stringify(written));
```
Comment (evilebottnawi) [inline src/writeFile.js]: Sync?

**Comment by deiga** [comment]
@evilebottnawi You're most likely right :P

Not even sure if this makes much sense to implement at all.
Some planning with other people would likely be a good idea.

**Comment by evilebottnawi** [comment]
@deiga i can do `cache` after some basic changes in this repo, thank you, can we close this here?

**Comment by deiga** [comment]
Sure, go ahead

**Pull request closed without merging**